blog/_src/posts/2016-06-27-tutorial-using-racket-s-ffi.scrbl

```diff
@@ -0,0 +1,335 @@
+#lang scribble/manual
+
+Title: Tutorial: Using Racket's FFI
+Date: 2016-06-27T16:22:11
+Tags: Racket, FFI, by Asumu Takikawa
+
+@(require scribble/example
+          scribble/racket
+          "../../utils/utils.rkt"
+          (for-syntax racket/base)
+          (for-label racket/base ffi/unsafe))
+@(define ev (make-base-eval))
+@(ev '(require racket/class))
+
+@(define-dummy _cairo_surface_t)
+@(define-dummy _cairo_t)
+@(define-dummy _cairo_line_cap_t)
+
+I've seen several people ask for a tutorial on Racket's foreign
+function interface (FFI), which allows you to dynamically load
+C libraries for use in Racket code. While I think the
+@hyperlink["http://docs.racket-lang.org/foreign/index.html"]{documentation}
+for the FFI is quite good, it is a lot of information to process and
+the overview examples may be tricky to run for a beginner.
+
+With that in mind, this blog post will provide a step-by-step tutorial
+for Racket's FFI that requires minimal setup. All that you will need to
+follow along is a copy of Racket and ideally a DrRacket window.
+
+<!-- more -->
+
+Before getting into the details, I wanted to note that the FFI library
+is based on the work of Eli Barzilay and Dmitry Orlovsky. They have
+a Scheme Workshop @hyperlink["http://www.ccs.neu.edu/racket/pubs/scheme04-bo.pdf"]{paper}
+that you can read if you're curious about the design.
+
+The tutorial will focus on using the @hyperlink["https://www.cairographics.org/"]{Cairo}
+graphics library, mainly because it comes bundled with Racket.
+
+To start, let's aim to reproduce the output of the "multi segment caps"
+C sample code on Cairo's
+@hyperlink["https://www.cairographics.org/samples/"]{samples page}:
+
+@codeblock[#:keep-lang-line? #f]|{
+#lang honu
+cairo_move_to (cr, 50.0, 75.0);
+cairo_line_to (cr, 200.0, 75.0);
+
+cairo_move_to (cr, 50.0, 125.0);
+cairo_line_to (cr, 200.0, 125.0);
+
+cairo_move_to (cr, 50.0, 175.0);
+cairo_line_to (cr, 200.0, 175.0);
+
+cairo_set_line_width (cr, 30.0);
+cairo_set_line_cap (cr, CAIRO_LINE_CAP_ROUND);
+cairo_stroke (cr);
+}|
+
+In order to actually draw this example to the screen, we will need a Cairo
+surface to draw on. So here is some boilerplate for you to execute before we
+actually play with the FFI:
+
+@examples[#:eval ev #:label #f
```

Comment:
While reading I wished there was a `#lang` declaration here. I see now why you didn't -- but if it's easy to footnote/margin-note I think that would be good to have.